django/db/backends/oracle/operations.py

```diff
@@ -334,8 +336,20 @@ def regex_lookup(self, lookup_type):
             match_option = "'i'"
         return 'REGEXP_LIKE(%%s, %%s, %s)' % match_option
 
-    def return_insert_id(self, field):
-        return 'RETURNING %s INTO %%s', (InsertVar(field),)
+    def return_insert_columns(self, fields):
+        if not fields:
+            return '', ()
+        columns = []
+        params = ()
+        into_params = []
+        for field in fields:
+            columns.append("%s.%s" % (self.quote_name(field.model._meta.db_table), self.quote_name(field.column)))
+            params += (InsertVar(field),)
+            into_params.append("%s")
```

Comment:
Simplify to the following?
```python
columns = ['%s.%s' % (self.quote_name(f.model._meta.db_table), self.quote_name(f.column)) for f in fields]
params = tuple(InsertVar(f) for f in fields)
```

Reply:
I guess simple is a bit subjective, yet I agree with you on this one. But, it's actually slower, since you need to loop over fields twice. Therefore, I'd like to keep this as is.

Reply:
I guess I find comprehensions more legible than loops like this, especially with single item tuple appends which are syntactically ugly. Comprehensions are fast with dedicated operations in bytecode, calling methods like `.append()` multiple times in the loop is slow.

In terms of performance, it should be negligible anyway as I expect this to be called once before the query executes, not per row returned, and typically I expect the number of fields to be very small...